controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -5775,13 +5776,51 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
               details: `The user doesnt have the permissions to perform this operation`,
             },
             logs: [],
+            count: '',
           };
         }
 
+        const analyticsRetention = await orgRepo.getFeature({
+          organizationId: authContext.organizationId,
+          featureId: 'analytics-retention',
+        });
+
+        const { dateRange } = validateDateRanges({
+          limit: analyticsRetention?.limit ?? 7,
+          dateRange: {
+            start: req.startDate,
+            end: req.endDate,
+          },
+        });
+
+        if (!dateRange) {
+          return {
+            response: {
+              code: EnumStatusCode.ERR,
+              details: 'Invalid date range',
+            },
+            logs: [],
+            count: '',
+          };
+        }
+
+        let limit = req.limit;
+        // check that the limit is less than the max option provided in the ui
+        if (limit > 50) {
```

Comment:
I'd check for 100 and throw an error if someone goes beyond otherwise it will hard to understand.

Reply:
Done, but kept the limit to 50 as that's the highest on the frontend

Reply:
ok